segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ContainerMetadataUpdateTransaction.java

```diff
@@ -481,6 +476,12 @@ private void updateMetadata(StreamSegmentMapping mapping, UpdateableSegmentMetad
 
         // Length must be at least StorageLength.
         metadata.setLength(Math.max(mapping.getLength(), metadata.getLength()));
+
+        // StartOffset must not exceed the last offset of the Segment.
+        if (metadata.getLength() > 0) {
+            metadata.setStartOffset(Math.min(mapping.getStartOffset(), metadata.getLength() - 1));
```

Comment:
The min call here implies that it is possible that `mapping.getStartOffset() > metadata.getLength() - 1`. How can we get to such a scenario?

Reply:
When the Segment Is empty (length ==0 ) and we truncate at offset 0.

Reply:
But if `length == 0`, then the condition to execute the if block is false and we will skip it, no?

Reply:
You're right. Then I guess my reasoning for putting it there was to have a failsafe in case upstream code did not do its job correctly. StartOffset should never exceed the Segment's length.